src/mender-update/daemon/states.cpp

```diff
@@ -103,15 +103,59 @@ void IdleState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) {
 	log::Debug("Entering Idle state");
 }
 
+SubmitInventoryState::SubmitInventoryState(
+	events::EventLoop &event_loop, int retry_interval_seconds, int retry_count) :
+	retry_(Retry {
```

Comment:
No need to create the temporary/intermediate `Retry` instance. `retry_()` in the CIL is a constructor call.